httpcore/_async/http11.py

```diff
@@ -168,12 +168,18 @@ async def _receive_response_body(self, request: Request) -> AsyncIterator[bytes]
             elif isinstance(event, (h11.EndOfMessage, h11.PAUSED)):
                 break
 
-    async def _receive_event(self, timeout: Optional[float] = None) -> H11Event:
+    async def _receive_event(
+        self, timeout: Optional[float] = None
+    ) -> Union[h11.Event, h11.PAUSED]:
         while True:
             with map_exceptions({h11.RemoteProtocolError: RemoteProtocolError}):
-                event = self._h11_state.next_event()
+                # The h11 type signature uses a private return type
+                event = cast(
+                    Union[h11.Event, h11.NEED_DATA, h11.PAUSED],
+                    self._h11_state.next_event(),
+                )
```

Comment:
So I need to do this on uvicorn as well? :disappointed:

Reply:
I'm not sure if you do following release of https://github.com/python-hyper/h11/pull/144 which updates that function's signature with narrowed return types.